### PR TITLE
Use @sap/project-piper-action-utils

### PR DESCRIPTION
**Pull request by kstiehl**
# Changes

* Use `@sap/project-piper-action-utils`
  * Update README.md so that the users adds `@sap` scope
* Reduce DevEnv Docker Image size by removing layers that are not necessary 
* stop downloading go environment for dev branch compilation

### CODE REVIEW

**Comment by fwilhe** [comment]
Hi @kstiehl 

Thanks for picking that up.

I tried to get the action running using the dev env container, but there seems to be an issue when running it the second time, cf

```
actions@ebaa63733870:~$ env INPUT_COMMAND=version INPUT_FLAGS='-v' node index.js
::debug::Looking up latest release
::debug::latest release was resolved to v1.146.0
Downloading latest release of piper
::debug::Downloading from https://api.github.com/repos/SAP/jenkins-library/releases/assets/38189772
::debug::isExplicit: 1.146.0
::debug::explicit? true
::debug::checking cache: /tmp/piper/1.146.0/x64
::debug::not found
::debug::Caching tool piper 1.146.0 x64
::debug::source file: /tmp/piper-download-3397203430939653
::debug::destination /tmp/piper/1.146.0/x64
::debug::destination file /tmp/piper/1.146.0/x64/piper
::debug::finished caching tool
::add-path::/tmp/piper/1.146.0/x64
[command]/tmp/piper/1.146.0/x64/piper version
piper-version:
    commit: "e794d4884bc9b8d8d8a8c798ce38184e7ee77856"
    tag: "<n/a>"
[command]/tmp/piper/1.146.0/x64/piper version -v
piper-version:
    commit: "e794d4884bc9b8d8d8a8c798ce38184e7ee77856"
    tag: "<n/a>"
actions@ebaa63733870:~$ env INPUT_COMMAND=version INPUT_FLAGS='-v' node index.js
::debug::Looking up latest release
::debug::latest release was resolved to v1.146.0
Downloading latest release of piper
::debug::Downloading from https://api.github.com/repos/SAP/jenkins-library/releases/assets/38189772
::debug::isExplicit: 1.146.0
::debug::explicit? true
::debug::checking cache: /tmp/piper/1.146.0/x64
::debug::Found tool in cache piper 1.146.0 x64
::error::Unable to locate executable file: piper. Please verify either the file path exists or the file can be found within a directory specified by the PATH environment variable. Also check the file mode to verify the file is executable.
```

Can you reproduce that? Looks like an issue if the binary is already in the cache, and then we don't use it from the cache but also don't have it in `PATH`.

**Comment by fwilhe** [comment]
Probably `Development.md` should be deleted in this PR, it only describes the feature to use non-released binaries which we remove here anyway.

**Comment by kstiehl** [comment]
> Can you reproduce that? Looks like an issue if the binary is already in the cache, and then we don't use it from the cache but also don't have it in `PATH`.

Yes good catch. I was always testing on machines where to runner cache was not working properly.
Should be fixed now.

> Probably Development.md should be deleted in this PR, it only describes the feature to use non-released binaries which we remove here anyway.

Done :+1:

File: startDevEnv.sh [hunk 1]
```diff
@@ -2,6 +2,5 @@
 
 npm install
 npm run package
```
Comment (fwilhe) [inline startDevEnv.sh:4]:
```suggestion
npm run prepare
```

This change is not strictly related to your pr, but seems required since at some point the default npm script name changed and we missed to reflect that here. For simplicity sake I would just add it here.

**Comment by fwilhe** [comment]
Please don't remove the mit license, it is in the repo because dependencies that are vendored into the repo (for the "compiled code") use it

**Comment by fwilhe** [comment]
looks like we need 8678c01303b4e6001d325b7118a0c2fce05ad689 and 6f1b8b14bd17cf55ba61fedcac75387d740c1a17 to enable gh package registry

Even with that changes, tests are failing in CI https://github.com/SAP/project-piper-action/runs/2828488673?check_suite_focus=true

"dist is up to date" should be fixed by committing the changes of `dist` folder after running `npm run all`

with the other checks, I assume the root cause is in the library, not sure, have not looked into details
